src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -288,6 +288,27 @@ fn check_contract_of(checker: &mut TypeChecker, args: &[SymbolicExpression], con
     Ok(TypeSignature::PrincipalType)
 }
 
+fn check_principal_of(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
+    check_argument_count(1, args)?;
+    checker.type_check_expects(&args[0], context, &BUFF_33)?;
+    Ok(TypeSignature::PrincipalType)
```

Comment:
Building on our last meeting, this function should be returning a PrincipalType wrapped in an Optional / Response instead.

Reply:
changed in 3907ed5